compiler/rustc_arena/src/lib.rs

```diff
@@ -362,6 +363,20 @@ unsafe impl<#[may_dangle] T> Drop for TypedArena<T> {
 
 unsafe impl<T: Send> Send for TypedArena<T> {}
 
+#[inline(always)]
+fn align_down(val: usize, align: usize) -> usize {
+    assert!(align.is_power_of_two());
+    val & !(align - 1)
+}
+
+#[inline(always)]
+fn align(val: usize, align: usize) -> usize {
```

Comment:
```suggestion
fn align_up(val: usize, align: usize) -> usize {
```
For symmetry with `align_down`.